src/codemodder/codemods/utils_mixin.py

```diff
@@ -17,24 +17,41 @@
 class NameResolutionMixin(MetadataDependent):
     METADATA_DEPENDENCIES: Tuple[Any, ...] = (ScopeProvider,)
 
+    def find_imported_name(self, node: cst.Name):
+        match self.find_single_assignment(node):
+            case ImportAssignment(
+                name=node.value,
+                node=(
+                    cst.Import(names=names) | cst.ImportFrom(names=names)
+                ) as import_node,
+            ) as assignment:
+                match names:
+                    case cst.ImportStar():
+                        return node.value
```

Comment:
This will never happen. You cannot attribute an assignment for a given to an import star because it can be ambiguous. Consider:
```python
from a import *
from b import *
foo()
```
Where does `foo()` come from?
If you need to convince yourself, try resolving the assignment for `Path` in this minimal example:
```python
from pathlib import *
Path("\")
```

Reply:
It can never happen but the type of `ImportFrom` allows `ImportStar` as the `name` so this is necessary for type correctness.

Reply:
Basically I could not figure out how else to restrict this using patterns in the match.